totalContributors.py

```diff
@@ -0,0 +1,66 @@
+import subprocess
+import pathlib
+import os.path
+import sys
+from datetime import datetime
+
+
+def add_log(dict, input):
+    for dataset in input:
+        name = dataset.split("\t")[1]
+        value = int(dataset.split("\t")[0])
+        if name in dict:
+            dict[name] += value
+        else:
+            dict[name] = value
+
+
+def retrieve_log(repo, version):
+    return subprocess.check_output(
+        ["git", "shortlog", "-s", version], universal_newlines=True, cwd=find_path(repo)
+    ).splitlines()
+
+
+def find_version(time, repo):
+    versions_list = subprocess.check_output(
+        ["git", "tag", "-l"], universal_newlines=True, cwd=find_path(repo)
+    ).splitlines()
+    for version in versions_list:
+        version_time = subprocess.check_output(
+            ["git", "log", "-1", "--format=%ai", version], universal_newlines=True, cwd=find_path(repo)
+        ).split()[0]
+        if datetime.strptime(version_time, "%Y-%m-%d") >= datetime.strptime(time, "%Y-%m-%d"):
+            return version
+    return version
+
+
+def find_path(repo):
+    return os.path.dirname(pathlib.Path().resolve()) + "/" + repo
+
+
+# extract git version and time
+if len(sys.argv) > 1:
```

Comment:
The `zulip_version`, which is the *previous* version for the starting point, should be printed out, in case the script user forgets to specify an argument. So they will know that it defaults to 4.3.